### PR TITLE
support predefined formats

### PR DESCRIPTION
**Pull request by lambrojos**
Hi I noticed that sometimes we have to use a predefined numFmt, instead of reading it from the workbook stryles. Thankfully ssf supports that

### CODE REVIEW

File: lib/worksheet.js [hunk 1]
```diff
@@ -236,7 +236,7 @@ XlsxStreamReaderWorkSheet.prototype._handleWorkSheetNode = function (nodeData) {
             if (self.workBook.hasFormatCodes) {
               var formatId = self.workBook.xfs[workingCell.attributes.s].attributes.numFmtId
               var format = self.workBook.formatCodes[formatId]
```
Comment (lambrojos) [inline lib/worksheet.js]: if we have to use a predefined format, this variable will turn out undefined

**Comment by DaSpawn** [comment]
I have merged the changes, but I am unsure what to do about this:
> if we have to use a predefined format, this variable will turn out undefined

File: lib/worksheet.js (hunk 1 above)
Comment (kikill95) [reply lib/worksheet.js]: Hm, maybe binary plus like "+formatId" will work faster? I am not sure

**Comment by DaSpawn** [comment]
please let me know if this looks ok, all tests pass.  I will push to npm if good

**Comment by lambrojos** [comment]
>I have merged the changes, but I am unsure what to do about this:

>>if we have to use a predefined format, this variable will turn out undefined

@DaSpawn sorry, that was bad english on my side
What I meant was that a cell can have a predefined format code and those  are not meant to be present in the workbook own formatcodes. I think that they are part of the xlsx standard but it looks like they are very poorly documented.. I found out about this topic here: https://social.msdn.microsoft.com/Forums/office/en-US/e27aaf16-b900-4654-8210-83c5774a179c/xlsx-numfmtid-predefined-id-14-doesnt-match?forum=oxmlsdk.
We've been lucky because ssf supports them out of the box https://github.com/SheetJS/ssf/blob/f335d310acd84a3d7cd5af3f0876a18a1c1887ac/bits/25_table.js

So, in this case `self.workBook.formatCodes[formatId]` will evaluate to undefined. So in this case we assume that `formatId` is a predefined format and we can pass it directly to ssf.

File: lib/worksheet.js (hunk 1 above)
Comment (lambrojos) [reply lib/worksheet.js]: Yes, maybe it is ... I didn't think about that.

Comment (lambrojos) [reply lib/worksheet.js]: https://jsperf.com/number-vs-plus-vs-toint-vs-tofloat/39 

looks like that bitwise not is faster?

**Comment by lambrojos** [comment]
@DaSpawn anyway I think that your implementation is a bit more defensive, we can publish to npm

**Comment by DaSpawn** [comment]
will do

**Pull request merged**